src/services/facilityService.ts

```diff
@@ -1,13 +1,48 @@
-import { facilities } from '../mockData/mockData';
 import { Facility } from '../typeDefs/gqlTypes';
+import {Facility as PrismaFacility, Contact as PrismaContact } from '@prisma/client';
+
+import prisma from '../db/client';
+
+type FacilityAndRelations = (PrismaFacility & {
+    contact: PrismaContact;
+})
+
+function convertPrismaToGraphQLFacility(input: FacilityAndRelations) {
+    return {
+        id: String(input.id),
+        nameJa: input.nameJa,
+        nameEn: input.nameEn,
+        contact: input.contact,
+        healthcareProfessionals: []
+    } as Facility;
+}
 
 // TODO: add a validation step for incoming parameters
-export const getFacilityById = (id: string) => {
-    const matchingFacility = facilities.find(
-        (location: Facility) => location.id === id
+export const getFacilityById = async (id: string) => {
+    const found = await prisma.facility.findUnique(
+        {where: {
+            id: parseInt(id)
+        },
+        include: {
+            contact: true
+        }}
     );
 
-    return matchingFacility;
+    return found ? convertPrismaToGraphQLFacility(found) : null;
 };
 
-export const getFacilities = () => facilities;
+export const getFacilities = async () => {
+    const facilities = await prisma.facility.findMany(
+        {include: {
+            contact: true
+        }}
+    );
+
+    const ret = Array<Facility>();
```

Comment:
nit: could we use more descriptive names? I couldn't figure out what `ret` was till I saw it was being returned. 

```suggestion
    const gqlFormattedFacilities = Array<Facility>();
```

Reply:
Yeah this is just very `golang` style 😎 I can adjust

Reply:
I'm going to try to find a happy medium between the golang `r` and `ret` and the Java `thisIsATemporaryVariableCounter` 

how about `gqlFacilites`

Reply:
💯 sounds great to me!